src/medCore/process/medAbstractProcess.cpp

```diff
@@ -12,18 +12,47 @@
 =========================================================================*/
 
 #include <medAbstractProcess.h>
+#include <medMessageController.h>
 #include <dtkCore/dtkAbstractObject.h>
 
 medAbstractProcess::medAbstractProcess(medAbstractProcess * parent) : dtkAbstractProcess(*new medAbstractProcessPrivate(this), parent)
 {
+    connect(this, SIGNAL(showError(QString, uint)), medMessageController::instance(), SLOT(showError(QString,uint)));
 }
 
 medAbstractProcess::medAbstractProcess(const medAbstractProcess& other) : dtkAbstractProcess(*new medAbstractProcessPrivate(*other.d_func()), other)
 {
-    
+    connect(this, SIGNAL(showError(QString, uint)), medMessageController::instance(), SLOT(showError(QString,uint)));
 }
 
 medAbstractProcess::~medAbstractProcess()
 {
 }
 
+void medAbstractProcess::displayVolumeError(QString id)
+{
+    QString error;
+    if ( !id.contains("3"))
+    {
+        error = "This toolbox is designed to be used with 3D volumes";
+    }
+    else
+    {
+        error = "Pixel type not yet implemented (" + id + ")";
+    }
+    qDebug() << description() + ": " + error;
+    emit showError(error, 3000);
+}
+
+int medAbstractProcess::isMeshTypeError(QString id)
```

Comment:
I agree with @LoicCadour.
I would avoid DTK_SUCCEED and DTK_FAILURE macros. They are meant to be used as indicators of the success of a process/job/task or whatever, not as general booleans.